Sources/PackageDescription/Target.swift

```diff
@@ -41,6 +41,12 @@ public final class Target {
         case `macro`
     }
 
+    /// A group a target belongs to that allows customizing access boundaries. A target is treated as
+    /// a client outside of the package if `excluded`, inside the package boundary if `package`.
+    public enum TargetGroup {
```

Comment:
This should have availability

Reply:
As discussed, that will require separate initializers per availability in a bunch of places; since static funcs that create targets are guarded with availability checks, it should be safe to leave this as is.